tests/test_camera.py

```diff
@@ -53,3 +57,56 @@ def read(self):
             video_source = VideoSource(camera_id=0)
         assert video_source._frames is None
         assert video_source.get_image() is not None
+
+
+class TestVideoStream(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.video = VideoSource(filename=VIDEO_PATH)
+        self.stream = VideoStream(video_source=self.video, fps=12.0)
+
+    def test_stop(self):
+        self.stream.stop()
+        self.assertTrue(self.stream._shutdown)
+
+    def test_extract_image(self):
+        img_tuple = self.video.get_image()
+        self.stream.frames.put(img_tuple, False)
+        imgs = self.stream.get_image()
+        assert type(imgs[0]) == np.ndarray
+
+    def test_frame_conditions(self):
+        self.stream.run()
+        self.assertTrue(self.stream.frames.full())
+        self.assertTrue(self.stream._shutdown)
+
+
+class TestVideoWriter(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.output_video_path = os.path.join(os.path.dirname(__file__), 'resources', 'test_writer.mp4')
+        self.videowriter = VideoWriter(path=self.output_video_path, fps=12.0, resolution=(40, 30))
+
+    def test_write(self):
+        input_video_path = VIDEO_PATH
+        input_video = cv2.VideoCapture(input_video_path)
+        output_video_path = self.output_video_path
+
+        while True:
+            ret, frame = input_video.read()
+            self.videowriter.write(frame)
+            if not ret:
+                break
+
+        input_video.release()
+        self.videowriter.release()
+        output_video = cv2.VideoCapture(output_video_path)
```

Comment:
To test whether the file is written, you open it with `VideoCapture` then assert that it indeed did. Instead, since `test_video.mp4` is the one being written again without modification, would it make sense to check that both files are the same and something hasn't modified/corrupted it in the process?